README.md
# Wave Theme

## General Information

*Wave* is a responsive Smarty theme for all OXID eShop editions.
This theme replaces deprecated [Flow theme](https://github.com/OXID-eSales/flow_theme).

## Branch Compatibility

* `b-7.0.x` is compatible with OXID eShop compilation `b-7.0.x`
* `b-6.5.x` is compatible with OXID eShop compilation `b-6.5.x`

## Installation
If you want to extend the theme you need to clone the repository (see section development) as some sources are ignored on composer installation. 


### Step 1: Get the source code

This step is only required if the theme is not delivered with the OXID distribution.
```
composer require oxid-esales/wave-theme
```

### Step 2: Activate theme 

Log into the admin panel, go to *Extensions → Themes → Wave* and press the *Activate* button. Clean the cache and off you go!

## Development

All *Wave* theme related CSS/Javascript files can be found in theme's ``build`` directory. To regenerate the theme's assets, the ``grunt`` tasks should be used. Please install ``grunt`` and run ``grunt's`` default task to regenerate all minimized ``css`` and ``js`` files:

To get the development files you need to clone the repository

	cd DOCUMENT_ROOT/Application/views/
	git clone -b BRANCH_NAME https://github.com/OXID-eSales/wave-theme.git wave
	
1. To use ``grunt``, ``npm`` is required. Check ``nodejs`` website for installation
instructions (https://nodejs.org/en/download/package-manager/). Example of
Installation on ubuntu system:

	```
	curl -sL https://deb.nodesource.com/setup_10.x | sudo -E bash -
	sudo apt-get install -y nodejs
	```

2. Go to "wave" theme's directory and install all related ``grunt`` plugins:

	```
    cd DOCUMENT_ROOT/Application/views/wave/
    npm install
    ```

3. Now its possible to regenerate "wave" theme assets by running ``grunt`` default
task while being in "wave" directory:

	```
	npm start
	```

## Contributing

If you are interested in contributing of some changes, Please read [CONTRIBUTING.md](CONTRIBUTING.md) file for more information.

## Issues

Please forward all issues to https://bugs.oxid-esales.com/ for the project **Wave Theme**.


CHANGELOG.md
# Change Log for OXID eSales Wave Theme

All notable changes to this project will be documented in this file.
The format is based on [Keep a Changelog](http://keepachangelog.com/)
and this project adheres to [Semantic Versioning](http://semver.org/).

## [3.0.1] - 2023-08-07

### Fixed
- Revert the ``user_account_newsletter`` block movement to the ``user_account`` template. Previously, it caused the newsletter subscription option to be unavailable for the registration form, eventually breaking the registration process.

### Changed
- License update

## [3.0.0] - 2023-04-20

### Removed
- RSS functionality

### Changed
- Use the recent minor version of Bootstrap 4

### Fixed
- In case of sAltImageUrl set, calculation of zoom image size is wrong [PR-123](https://github.com/OXID-eSales/wave-theme/pull/123)

## [2.0.0] - 2021-07-07

### Added
- File ``build/scss/custom/styles.scss`` for easier splitting of custom and original styles [PR-96](https://github.com/OXID-eSales/wave-theme/pull/96)
- File ``build/scss/custom/variables.scss`` for easier overwriting of default original variable values [PR-96](https://github.com/OXID-eSales/wave-theme/pull/96)
- Configuration of instagram channel is added [PR-117](https://github.com/OXID-eSales/wave-theme/pull/117)

### Changed
- Updated the version of bootstrap from 4.1.3 to 4.3.1 [PR-75](https://github.com/OXID-eSales/wave-theme/pull/75) 
- BC Break: block in ``tpl/email/plain/ordershopped.tpl`` renamed from ``email_html_ordershipped_shipmenttrackingurl`` to ``email_plain_ordershipped_shipmenttrackingurl`` [PR-73](https://github.com/OXID-eSales/wave-theme/pull/73)
- Decoded ``oxconfig`` insert queries in ``setup.sql``
- Handle Google analytics scripts with oxscript [PR-87](https://github.com/OXID-eSales/wave-theme/pull/87)
- In admin settings assigned 'e-mail-logo' to display group 'emails' [PR-97](https://github.com/OXID-eSales/wave-theme/pull/97)
- Removed checkout step 5 from the progress bar [PR-77](https://github.com/OXID-eSales/wave-theme/pull/77)
- Simplify checkout step 2 (user registration) [PR-68](https://github.com/OXID-eSales/wave-theme/pull/68)
  - Registration / no registration form is displayed all the time, right next to the "Already a customer" block
  - "Purchase without registration" checkbox added, to differ between registration / no registration checkouts
  - "Subscribe to the newsletter" and "Drop us a message" inputs are extracted to separate block

### Fixed
- Improved readability of setup.sql and theme.php [PR-97](https://github.com/OXID-eSales/wave-theme/pull/97)
- Remove dependency on a system wide installed grunt-cli [PR-102](https://github.com/OXID-eSales/wave-theme/pull/102)

### Removed
- Google+ related settings together with frontend and email template blocks [PR-81](https://github.com/OXID-eSales/wave-theme/pull/81)
- Credit Card related blocks
- Remove Betanote
- Suggest (Recommend Product) feature:
    - Templates:
        - `email/html/suggest.tpl`
        - `email/plain/suggest.tpl`
        - `form/suggest.tpl`
        - `page/info/suggest.tpl`
    - Link in: `page/details/inc/productmania.tpl`
- News feature
- `setup.sql` file support and file itself
- Remove `ajax_econda_recommendations.tpl` use econda module instead

## [1.8.0] - 2022-12-02

### Changed
- Switch from using Google Fonts API to self-hosted font files storage

### Fixed
- Bootstrap version in `package.json` set to v4.3.1

## [1.7.0] - 2022-11-15

### Added
- Template blocks on login form:
  - `checkout_options_login_passwordfield` in `login.tpl`
  - `checkout_options_loginaccount_passwordfield` in `login_account.tpl`
  - `checkout_options_loginaccount_submitbutton` in `login_account.tpl`

## [1.6.2] - 2022-07-18

### Fixed
- CONTRIBUTING link

## [1.6.1] - 2021-07-07

### Fixed
- Use getTitle from controller, but not actCategory, which is not always available [PR-120](https://github.com/OXID-eSales/wave-theme/pull/120)
- Fix multilang in errors_modal.tpl [PR-119](https://github.com/OXID-eSales/wave-theme/pull/119)
- Add missing actions.tpl for Actions widget [PR-121](https://github.com/OXID-eSales/wave-theme/pull/121)

## [1.6.0] - 2021-03-12

### Added
- Add the autocomplete HTML attributes to most HTML input fields [PR-107](https://github.com/OXID-eSales/wave-theme/pull/107)
- Sku and offer url microdata added [PR-108](https://github.com/OXID-eSales/wave-theme/pull/108)
- New blocks:
  - ``captcha_form`` in:
    - ``tpl/form/forgotpwd_email.tpl`` [Flow-PR-163](https://github.com/OXID-eSales/flow_theme/pull/163)
    - ``tpl/form/newsletter.tpl`` [Flow-PR-163](https://github.com/OXID-eSales/flow_theme/pull/163)
  - in ``tpl/widget/product/list``:
    - ``widget_product_list`` [PR-113](https://github.com/OXID-eSales/wave-theme/pull/113)
    - ``widget_product_list_head`` [PR-113](https://github.com/OXID-eSales/wave-theme/pull/113)
    - ``widget_product_list_item`` [PR-113](https://github.com/OXID-eSales/wave-theme/pull/113)
  - ``salutation_options`` in ``tpl/form/fieldset/salutation.tpl`` [PR-112](https://github.com/OXID-eSales/wave-theme/pull/112)
  - ``page_account_forgot_email_sent`` in ``tpl/page/account/forgotpwd.tpl`` [PR-115](https://github.com/OXID-eSales/wave-theme/pull/115)
  - in ``tpl/email/html/order_cust.tpl`` [PR-116](https://github.com/OXID-eSales/wave-theme/pull/116):
    - ``email_html_order_cust_basketitem_title``
    - ``email_html_order_cust_basketitem_sellist``
    - ``email_html_order_cust_basketitem_persparams``
    - ``email_html_order_cust_basketitem_artnum``
    - ``email_html_order_cust_basketitem_unitprice``
    - ``email_html_order_cust_basketitem_discounts``
    - ``email_html_order_cust_basketitem_orderinfo``
    - ``email_html_order_cust_basketitem_amount``
    - ``email_html_order_cust_basketitem_vat``
    - ``email_html_order_cust_basketitem_price``
  - in ``tpl/email/plain/order_cust.tpl`` [PR-116](https://github.com/OXID-eSales/wave-theme/pull/116):
    - ``email_plain_order_cust_title``

theme.php
<?php

/**
 * This file is part of OXID eSales Wave theme.
 *
 * OXID eSales Wave theme is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * OXID eSales Wave theme is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with OXID eSales Wave theme.  If not, see <http://www.gnu.org/licenses/>.
 *
 * @link      http://www.oxid-esales.com
 * @copyright (C) OXID eSales AG 2003-2016
 */

declare(strict_types=1);

/**
 * Theme Information
 */
$aTheme = [
    'id' => 'wave',
    'title' => 'Wave',
    'description' => 'Wave is OXID`s official responsive theme based on the CSS framework Bootstrap 4.',
    'thumbnail' => 'theme.jpg',
    'version' => '3.0.1',
    'author' => '<a href="https://www.oxid-esales.com" title="OXID eSales AG">OXID eSales AG</a>',
    'settings' => [
        //Group display:
        [
            'group' => 'display',
            'name' => 'blFullwidthLayout',
            'type' => 'bool',
            'value' => 0,
            'position' => 1001,
        ],
        [
            'group' => 'display',
            'name' => 'blHomeLink',
            'type' => 'bool',
            'value' => 1,
            'position' => 1002,
        ],
        [
            'group' => 'display',
            'name' => 'bl_showManufacturerSlider',
            'type' => 'bool',
            'value' => 1,
            'position' => 1003,
        ],
        [
            'group' => 'display',
            'name' => 'sDefaultListDisplayType',
            'type' => 'select',
            'value' => 'infogrid',
            'constraints' => 'infogrid|line|grid',
            'position' => 1004,
        ],
        [
            'group' => 'display',
            'name' => 'blShowListDisplayType',
            'type' => 'bool',
            'value' => 1,
            'position' => 1005,
        ],
        [
            'group' => 'display',
            'name' => 'sStartPageListDisplayType',
            'type' => 'select',
            'value' => 'grid',
            'constraints' => 'infogrid|line|grid',
            'position' => 1006,
        ],
        [
            'group' => 'display',
            'name' => 'aNrofCatArticles',
            'type' => 'arr',
            'value' => ['10', '20', '50', '100'],
            'position' => 1007,
        ],
        [
            'group' => 'display',
            'name' => 'aNrofCatArticlesInGrid',
            'type' => 'arr',
            'value' => ['12', '16', '24', '32'],
            'position' => 1008,
        ],
        [
            'group' => 'display',
            'name' => 'iNewBasketItemMessage',
            'type' => 'select',
            'value' => 'str',
            'constraints' => '0|1|2|3',
            'position' => 1009,
        ],
        [
            'group' => 'display',
            'name' => 'blShowFinalStep',
            'type' => 'bool',
            'value' => 1,
            'position' => 1010,
        ],
        [
            'group' => 'display',
            'name' => 'blShowBirthdayFields',
            'type' => 'bool',
            'value' => 1,
            'position' => 1011,
        ],
        //Group logo:
        [
            'group' => 'logo',
            'name' => 'sLogoFile',
            'type' => 'str',
            'value' => 'logo_oxid.png',
            'position' => 1101,
        ],
        [
            'group' => 'logo',
            'name' => 'sLogoWidth',
            'type' => 'str',
            'value' => '200',
            'position' => 1102,
        ],
        [
            'group' => 'logo',
            'name' => 'sLogoHeight',
            'type' => 'str',
            'value' => '38',
            'position' => 1103,
        ],
        //Group favicon:
        [
            'group' => 'favicons',
            'name' => 'sFaviconFile',
            'type' => 'str',
            'value' => 'favicon.ico',
            'position' => 1201,
        ],
        [
            'group' => 'favicons',
            'name' => 'sFavicon16File',
            'type' => 'str',
            'value' => 'favicon_16x16.png',
            'position' => 1202,
        ],
        [
            'group' => 'favicons',
            'name' => 'sFavicon32File',
            'type' => 'str',
            'value' => 'favicon_32x32.png',
            'position' => 1203,
        ],
        [
            'group' => 'favicons',
            'name' => 'sFavicon48File',
            'type' => 'str',
            'value' => 'favicon_48x48.png',
            'position' => 1204,
        ],
        [
            'group' => 'favicons',
            'name' => 'sFavicon64File',
            'type' => 'str',
            'value' => 'favicon_64x64.png',
            'position' => 1205,
        ],
        [
            'group' => 'favicons',
            'name' => 'sFavicon128File',
            'type' => 'str',
            'value' => 'favicon_128x128.png',
            'position' => 1206,
        ],
        [
            'group' => 'favicons',
            'name' => 'sFavicon512File',
            'type' => 'str',
            'value' => 'favicon_512x512.png',
            'position' => 1207,
        ],
        [
            'group' => 'favicons',
            'name' => 'sFaviconMSTileColor',
            'type' => 'str',
            'value' => '#D83434',
            'position' => 1208,
        ],
        //Group images:
        [
            'group' => 'images',
            'name' => 'sIconsize',
            'type' => 'str',
            'value' => '87*87',
            'position' => 1301,
        ],
        [
            'group' => 'images',
            'name' => 'sThumbnailsize',
            'type' => 'str',
            'value' => '390*245',
            'position' => 1302,
        ],
        [
            'group' => 'images',
            'name' => 'aDetailImageSizes',
            'type' => 'aarr',
            'value' => [
                'oxpic1' => '540*340',
                'oxpic2' => '540*340',
                'oxpic3' => '540*340',
                'oxpic4' => '540*340',
                'oxpic5' => '540*340',
                'oxpic6' => '540*340',
                'oxpic7' => '540*340',
                'oxpic8' => '540*340',
                'oxpic9' => '540*340',
                'oxpic10' => '540*340',
                'oxpic11' => '540*340',
                'oxpic12' => '540*340',
            ],
            'position' => 1303,
        ],
        [
            'group' => 'images',
            'name' => 'sZoomImageSize',
            'type' => 'str',
            'value' => '665*665',
            'position' => 1304,
        ],
        [
            'group' => 'images',
            'name' => 'sCatIconsize',
            'type' => 'str',
            'value' => '168*100',
            'position' => 1305,
        ],
        [
            'group' => 'images',
            'name' => 'sCatPromotionsize',
            'type' => 'str',
            'value' => '370*107',
            'position' => 1306,
        ],
        [
            'group' => 'images',
            'name' => 'sCatThumbnailsize',
            'type' => 'str',
            'value' => '1140*250',
            'position' => 1307,
        ],
        [
            'group' => 'images',
            'name' => 'sManufacturerIconsize',
            'type' => 'str',
            'value' => '100*100',
            'position' => 1308,
        ],
        [
            'group' => 'images',
            'name' => 'blSliderShowImageCaption',
            'type' => 'bool',
            'value' => 1,
            'position' => 1309,
        ],
        //Group background:
        [
            'group' => 'background',
            'name' => 'sBackgroundColor',
            'type' => 'str',
            'value' => '#CCEBF5',
            'position' => 1401,
        ],
        [
            'group' => 'background',
            'name' => 'blUseBackground',
            'type' => 'bool',
            'value' => 0,
            'position' => 1402,
        ],
        [
            'group' => 'background',
            'name' => 'sBackgroundPath',
            'type' => 'str',
            'value' => 'mein_hintergrundbild.jpg',
            'position' => 1403,
        ],
        [
            'group' => 'background',
            'name' => 'sBackgroundSize',
            'type' => 'select',
            'value' => 'cover',
            'constraints' => 'cover|contain|normal',
            'position' => 1404,
        ],
        [
            'group' => 'background',
            'name' => 'sBackgroundPosHorizontal',
            'type' => 'select',
            'value' => 'center',
            'constraints' => 'left|right|center',
            'position' => 1405,
        ],
        [
            'group' => 'background',
            'name' => 'sBackgroundPosVertical',
            'type' => 'select',
            'value' => 'top',
            'constraints' => 'top|bottom|center',
            'position' => 1406,
        ],
        [
            'group' => 'background',
            'name' => 'sBackgroundRepeat',
            'type' => 'select',
            'value' => 'no-repeat',
            'constraints' => 'no-repeat|repeat-x|repeat-y|repeat',
            'position' => 1407,
        ],
        [
            'group' => 'background',
            'name' => 'blBackgroundAttachment',
            'type' => 'bool',
            'value' => 1,
            'position' => 1408,
        ],
        //Group features:
        [
            'group' => 'features',
            'name' => 'bl_showCompareList',
            'type' => 'bool',
            'value' => 1,
            'position' => 1501,
        ],
        [
            'group' => 'features',
            'name' => 'bl_showPriceAlarm',
            'type' => 'bool',
            'value' => 1,
            'position' => 1502,
        ],
        [
            'group' => 'features',
            'name' => 'bl_showWishlist',
            'type' => 'bool',
            'value' => 1,
            'position' => 1503,
        ],
        [
            'group' => 'features',
            'name' => 'bl_showListmania',
            'type' => 'bool',
            'value' => 1,
            'position' => 1504,
        ],
        [
            'group' => 'features',
            'name' => 'bl_showVouchers',
            'type' => 'bool',
            'value' => 1,
            'position' => 1505,
        ],
        [
            'group' => 'features',
            'name' => 'bl_showGiftWrapping',
            'type' => 'bool',
            'value' => 1,
            'position' => 1506,
        ],
        //Group footer:
        [
            'group' => 'footer',
            'name' => 'blFooterShowHelp',
            'type' => 'bool',
            'value' => 1,
            'position' => 1601,
        ],
        [
            'group' => 'footer',
            'name' => 'blFooterShowLinks',
            'type' => 'bool',
            'value' => 1,
            'position' => 1602,
        ],
        [
            'group' => 'footer',
            'name' => 'blFooterShowNewsletter',
            'type' => 'bool',
            'value' => 1,
            'position' => 1604,
        ],
        [
            'group' => 'footer',
            'name' => 'blFooterShowNewsletterForm',
            'type' => 'bool',
            'value' => 1,
            'position' => 1605,
        ],
        [
            'group' => 'footer',
            'name' => 'sFacebookUrl',
            'type' => 'str',
            'value' => 'https://www.facebook.com',
            'position' => 1606,
        ],
        [
            'group' => 'footer',
            'name' => 'sTwitterUrl',
            'type' => 'str',
            'value' => 'https://twitter.com',
            'position' => 1607,
        ],
        [
            'group' => 'footer',
            'name' => 'sYouTubeUrl',
            'type' => 'str',
            'value' => 'https://www.youtube.com',
            'position' => 1608,
        ],
        [
            'group' => 'footer',
            'name' => 'sBlogUrl',
            'type' => 'str',
            'value' => 'https://wordpress.org',
            'position' => 1609,
        ],
        [
            'group' => 'footer',
            'name' => 'sInstagramUrl',
            'type' => 'str',
            'value' => 'https://www.instagram.com/',
            'position' => 1610,
        ],
        //Group contact:
        [
            'group' => 'contact',
            'name' => 'sGoogleMapsAddr',
            'type' => 'str',
            'value' => 'OXID eSales AG, Bertoldstraße 48, 79098 Freiburg',
            'position' => 1701,
        ],
        //Group emails:
        [
            'group' => 'emails',
            'name' => 'sEmailLogo',
            'type' => 'str',
            'value' => 'logo_email.png',
            'position' => 1801,
        ],
        [
            'group' => 'emails',
            'name' => 'blEmailsShowProductPictures',
            'type' => 'bool',
            'value' => 0,
            'position' => 1802,
        ],
        //Group googleanalytics:
        [
            'group' => 'googleanalytics',
            'name' => 'sGATrackingId',
            'type' => 'str',
            'value' => '',
            'position' => 1901,
        ],
        [
            'group' => 'googleanalytics',
            'name' => 'blUseGAEcommerceTracking',
            'type' => 'bool',
            'value' => 1,
            'position' => 1902,
        ],
        [
            'group' => 'googleanalytics',
            'name' => 'blUseGAPageTracker',
            'type' => 'bool',
            'value' => 1,
            'position' => 1903,
        ],
        [
            'group' => 'googleanalytics',
            'name' => 'blGAAnonymizeIPs',
            'type' => 'bool',
            'value' => 1,
            'position' => 1904,
        ],
        //Group googlets:
        [
            'group' => 'googlets',
            'name' => 'blUseGoogleTS',
            'type' => 'bool',
            'value' => 0,
            'position' => 2001,
        ],
        [
            'group' => 'googlets',
            'name' => 'sGoogleVendorId',
            'type' => 'str',
            'value' => '',
            'position' => 2002,
        ],
        [
            'group' => 'googlets',
            'name' => 'sPageLanguage',
            'type' => 'str',
            'value' => 'de_DE',
            'position' => 2003,
        ],
        [
            'group' => 'googlets',
            'name' => 'sGoogleShoppingAccountId',
            'type' => 'str',
            'value' => '',
            'position' => 2004,
        ],
        [
            'group' => 'googlets',
            'name' => 'sShippingDaysOnStock',
            'type' => 'str',
            'value' => '3',
            'position' => 2005,
        ],
        [
            'group' => 'googlets',
            'name' => 'sShippingDaysNotOnStock',
            'type' => 'str',
            'value' => '5',
            'position' => 2006,
        ],
        [
            'group' => 'googlets',
            'name' => 'sDeliveryDaysOnStock',
            'type' => 'str',
            'value' => '6',
            'position' => 2007,
        ],
        [
            'group' => 'googlets',
            'name' => 'sDeliveryDaysNotOnStock',
            'type' => 'str',
            'value' => '14',
            'position' => 2008,
        ],
        [
            'group' => 'googlets',
            'name' => 'sShoppingLanguage',
            'type' => 'str',
            'value' => 'de',
            'position' => 2009,
        ],
        [
            'group' => 'googlets',
            'name' => 'sShoppingCountry',
            'type' => 'str',
            'value' => 'DE',
            'position' => 2010,
        ],
    ],
];
